blog/zk-snark-pallet.md

```diff
@@ -0,0 +1,263 @@
+# Zk-SNARKs with Substrate (Part 3)
+
+This is the last part of a series of articles about zk-SNARKs. If you haven't read the previous ones (Part 1, Part 2), I encourage you to do so now.
+
+In our first article, we defined Bob's problem. He creates a contest, where the first person who solves the equation: 
+
+<center>
+
+$$ x^2+3=12 $$
+
+</center>
+
+will be able to join his Bright Coders union. As you probably remember, Alice was one of his friends who knows the solution. She was only afraid that revealing it loudly, could encourage others to first claim the vacancy. That's why she decided to use zk-SNARKs.
+
+Until this moment, everything that we did assumed Bob and Alice are in the same place. Alice decides to create proof because she doesn't want to reveal the solution to anyone. Bob verified it in front of her, so Alice was certain that she was the first who solve the puzzle. But what if they are not in the same place? What guarantee will have Alice, that her solution was verified first? This problem can be easily solved, if we could move it to the blockchain! Information about the winner will be known to everyone, and the verification process will be more transparent.
```

Comment:
To bym przerobiła na czas przeszły: Alice decides to create proof because she doesn't want to reveal the solution to anyone.

Reply:
What guarantee will have Alice - wydaje mi się, że powinno być: What guarantee will Alice have...

Reply:
But what if they are not in the same place? - Wydaje mi się, że to przypuszczenie i dałabym [..] if they weren't in the same place?

Reply:
Przecinek niepotrzebny, bo nie zaczynamy od if: This problem can be easily solved, if we could move it to the blockchain!